src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscores.Fixer.cs

```diff
@@ -35,15 +35,15 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
             string title = MicrosoftCodeQualityAnalyzersResources.IdentifiersShouldNotContainUnderscoresTitle;
             context.RegisterCodeFix(new MyCodeAction(title,
-                                        async ct => await RenameWithoutUnderscores(context.Document, symbol, ct).ConfigureAwait(false),
+                                        async ct => await RenameWithoutUnderscoresAsync(context.Document, symbol, ct).ConfigureAwait(false),
                                         equivalenceKey: title),
                                     context.Diagnostics);
         }
 
-        private static async Task<Document> RenameWithoutUnderscores(Document document, ISymbol symbol, CancellationToken cancellationToken)
+        private static async Task<Document> RenameWithoutUnderscoresAsync(Document document, ISymbol symbol, CancellationToken cancellationToken)
         {
             Solution newSolution = await Renamer.RenameSymbolAsync(document.Project.Solution, symbol, symbol.Name.Replace("_", ""), null, cancellationToken).ConfigureAwait(false);
```

Comment:
Not sure if `symbol.Name.Replace("_", "")` is a good suggestion. For example, if I had something like `This_is_a_method`, it is likely, I want `ThisIsAMethod` or something else, but certainly not `Thisisamethod`.

Reply:
@mavasani
Does something like this makes sense?
It keeps all the letter as-is except that all letters after `_` will be capitalized.

```csharp
        private static string GetNewName(string name)
        {
            var builder = new StringBuilder();
            bool isNextUpperCase = false;
            foreach (char c in name)
            {
                if (c == '_')
                {
                    isNextUpperCase = true;
                    continue;
                }
                builder.Append(isNextUpperCase ? char.ToUpper(c, CultureInfo.InvariantCulture) : c);
                isNextUpperCase = false;
            }
            return builder.ToString();
        }
```

Reply:
Sounds reasonable - might want to rename `isNextUpperCase` to `isPreviousUnderscore` for clarity.